packages/ui/components/ui/tx/index.tsx

```diff
@@ -2,60 +2,92 @@ import { TransactionView } from '@buf/penumbra-zone_penumbra.bufbuild_es/penumbr
+  assetId?: AssetId;
+}) => Promise<Metadata | undefined>;
 
-  // Request the fee 'Metadata' and construct a 'ValueView' object
-  const chainId = txv.bodyView.transactionParameters.chainId;
-  const assetId = txv.bodyView.transactionParameters.fee.assetId;
-  const feeAssetMetadata = getFeeAssetMetadataOrDefault(chainId, assetId);
-  const feeValueView = feeAssetMetadata
-    ? new ValueView({
-        valueView: {
-          case: 'knownAssetId',
-          value: {
-            amount: txv.bodyView.transactionParameters.fee.amount,
-            metadata: feeAssetMetadata,
-          },
-        },
+// Uses supplied metadata fetcher to see if it can augment fee ValueView with metadata
+const useFeeMetadata = (txv: TransactionView, getMetadata: MetadataFetchFn) => {
+  const amount = txv.bodyView?.transactionParameters?.fee?.amount;
+  const [feeValueView, setFeeValueView] = useState<ValueView>(
+    new ValueView({
+      valueView: {
+        case: 'unknownAssetId',
+        value: { amount },
+      },
+    }),
+  );
+
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+  const [error, setError] = useState<unknown>();
+
+  useEffect(() => {
+    const chainId = txv.bodyView?.transactionParameters?.chainId;
+    const assetId = txv.bodyView?.transactionParameters?.fee?.assetId;
+    setIsLoading(true);
+    void getMetadata({ chainId, assetId })
+      .then(metadata => {
+        if (metadata) {
+          const feeValueView = new ValueView({
+            valueView: {
+              case: 'knownAssetId',
+              value: { amount, metadata },
+            },
+          });
+          setFeeValueView(feeValueView);
+        }
+        setIsLoading(false);
       })
-    : new ValueView({
-        valueView: {
-          case: 'unknownAssetId',
-          value: { amount: txv.bodyView.transactionParameters.fee.amount },
-        },
-      });
+      .catch((e: unknown) => setError(e));
+  }, [txv, getMetadata, setFeeValueView]);
+
+  return { feeValueView, isLoading, error };
+};
+
+export const TransactionViewComponent = ({
+  txv,
+  metadataFetcher,
+}: {
+  txv: TransactionView;
+  metadataFetcher: MetadataFetchFn;
+}) => {
+  const { feeValueView, isLoading, error } = useFeeMetadata(txv, metadataFetcher);
 
   return (
     <div className='flex flex-col gap-8'>
-      {txv.bodyView.memoView?.memoView && <MemoViewComponent memo={txv.bodyView.memoView} />}
+      {txv.bodyView?.memoView?.memoView && <MemoViewComponent memo={txv.bodyView.memoView} />}
       <ViewSection heading='Actions'>
-        {txv.bodyView.actionViews.map((av, i) => (
+        {txv.bodyView?.actionViews.map((av, i) => (
           <ActionViewComponent av={av} feeValueView={feeValueView} key={i} />
         ))}
       </ViewSection>
       <ViewSection heading='Parameters'>
         <ViewBox
           label='Transaction Fee'
           visibleContent={
-            <div className='font-mono'>
+            <div className='flex items-center gap-2'>
```

Comment:
think this fixes https://github.com/penumbra-zone/web/issues/1474?

Reply:
Indeed it does 👍